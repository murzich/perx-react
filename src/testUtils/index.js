// eslint-disable-next-line import/prefer-default-export
export { default as renderWithRedux } from './renderWithRedux';
