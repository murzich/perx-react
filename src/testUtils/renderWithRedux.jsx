// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../state/reducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

export const rerenderWithRedux = rerender => (
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) => ({
  ...rerender(<Provider store={store}>{ui}</Provider>),
  store,
});
