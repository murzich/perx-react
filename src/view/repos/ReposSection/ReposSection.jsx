import React from 'react';
import { Header } from 'semantic-ui-react';
import ReposContainer from '../ReposContainer';

const ReposSection = () => (
  <section>
    <Header as="h2">Repositories</Header>
    <ReposContainer />
  </section>
);

export default ReposSection;
