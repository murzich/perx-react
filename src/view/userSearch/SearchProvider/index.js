import SearchProvider from './SearchProvider';

export default SearchProvider;
