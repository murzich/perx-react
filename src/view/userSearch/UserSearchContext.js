import { createContext } from 'react';

const UserSearchContext = createContext();

export default UserSearchContext;
