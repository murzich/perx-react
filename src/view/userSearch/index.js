import SearchForm from './SearchForm';
import SearchProvider from './SearchProvider';
import UserSearchContext from './UserSearchContext';

export { UserSearchContext, SearchProvider };

export default SearchForm;
