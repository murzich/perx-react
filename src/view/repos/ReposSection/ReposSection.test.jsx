import { cleanup } from '@testing-library/react';
import React from 'react';
import { renderWithRedux } from '../../../testUtils';
import { UserSearchContext } from '../../userSearch';
import ReposSection from './ReposSection';

describe('ReposSection', () => {
  afterEach(cleanup);

  const testUser = 'testUser';

  it('should match snapshot', () => {
    const { asFragment } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <ReposSection />
      </UserSearchContext.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
