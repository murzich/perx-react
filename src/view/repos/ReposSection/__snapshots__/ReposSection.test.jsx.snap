// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposSection should match snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="ui header"
    >
      Repositories
    </h2>
    <div
      class="ui divided relaxed list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            Enter user name to search for his repositories
          </div>
          <div
            class="description"
          >
            There should be a list of all his repositories with public access.
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
